frontend/src/tests/application-tests.js

```diff
@@ -1,24 +1,391 @@
+
+        it.skip(
+            "Submitting an application for a posting that does not have any positions or position preferences" +
+                " responds in error",
+            async () => {
+                // Create a new posting
+                const posting = {
+                    name: "CSC343F TA",
+                    intro_text: "Testing posting for CSC343F",
+                    open_date: "2021/04/01",
+                    close_date: "2021/05/01",
+                    availability: "open",
+                };
+                let resp = await apiPOST("/admin/postings", {
+                    ...posting,
+                    session_id: session.id,
+                });
+                expect(resp).toHaveStatus("success");
+                checkPropTypes(postingPropTypes, resp.payload);
+                expect(resp.payload.id).not.toBeNull();
+                Object.assign(posting, resp.payload);
+
+                await switchToTaUser();
+
+                // Submit an application without linking a position to the posting
+
+                // ISSUE: Even though we should not be able to submit an application without a position linked to
+                // it, this POST request successfully creates an application with position_preferences being an
+                // empty object.
+
+                const correctApplication = {
+                    answers: {
+                        ...applicantInfo,
+                        position_preferences: {
+                            [position.position_code]: HIGH_PREFERENCE,
+                        },
+                    },
+                };
+
+                resp = await apiPOST(
+                    `/public/postings/${posting.url_token}/submit`,
+                    correctApplication,
+                    true
+                );
+                expect(resp).toHaveStatus("error");
+
+                await restoreDefaultUser();
+
+                // Link seeded position to the new posting
+                const postingPosition = {
+                    num_positions: 10,
+                    hours: 68,
+                };
+                resp = await apiPOST(
+                    `/admin/postings/${posting.id}/posting_positions`,
+                    { ...postingPosition, position_id: position.id }
+                );
+                expect(resp).toHaveStatus("success");
+                Object.assign(postingPosition, resp.payload);
+
+                // Submit an application without position_preferences
+                const applicationWithoutPositionPref = {
+                    answers: {
+                        ...applicantInfo,
+                    },
+                };
+                resp = await apiPOST(
+                    `/public/postings/${posting.url_token}/submit`,
+                    applicationWithoutPositionPref,
+                    true
+                );
+                expect(resp).toHaveStatus("error");
```

Comment:
I remember from the meeting yesterday there were some updates you still have to push, right?